vcsclient/fs.go

```diff
@@ -200,6 +197,28 @@ func GetFileWithOptions(fs vfs.FileSystem, path string, opt GetFileOptions) (*Fi
 	return &fwr, nil
 }
 
+// readDir uses the passed vfs.FileSystem to read from starting at the base path. If
+// shouldRecurse is set to true, it will recurse into all sub-folders and return the
+// full sub-tree.
+func readDir(fs vfs.FileSystem, base string, shouldRecurse bool) ([]*TreeEntry, error) {
```

Comment:
Totally. Unfortunately, [`vfs.FileSystem`](https://sourcegraph.com/golang.org/x/tools@7d75e8b219c3efda2d08ae38acd6b42f8da4f5f9/.GoPackage/golang.org/x/tools/godoc/vfs/.def/FileSystem) can not be converted [`fs.FileSystem`](https://sourcegraph.com/github.com/kr/fs@2788f0dbd16903de03cb8186e5c7d97b69ad387b/.tree/filesystem.go#def=/github.com/kr/fs@2788f0dbd16903de03cb8186e5c7d97b69ad387b/.GoPackage/github.com/kr/fs/.def/FileSystem&startbyte=126&endbyte=774) because it is lacking the [`Join`](https://sourcegraph.com/github.com/kr/fs@2788f0dbd16903de03cb8186e5c7d97b69ad387b/.tree/filesystem.go#startline=26&endline=26) method - bummer!